tests/manage/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py

```diff
@@ -0,0 +1,202 @@
+        logging.info('total space  before expansion = {}'.format(str(total_space_b4_expansion)))
+
+        logging.info("Perform some IO operations...")
+
+        pvc_objs = multi_pvc_factory(
+            size=self.pvc_size, num_of_pvc=self.num_of_pvcs
+        )
+
+        pod_objs = []
+
+        with ThreadPoolExecutor(max_workers=self.num_of_pvcs) as executor:
+            for pvc_obj in pvc_objs:
+                executor.submit(pod_objs.append, (pod_factory(pvc=pvc_obj)))
+
+        with ThreadPoolExecutor(max_workers=self.num_of_pvcs-1) as executor:
+            for pod_obj in pod_objs:
+                pod_io_task = executor.submit(
+                    pod_obj.run_io, storage_type='fs', size=4, jobs=10, io_direction='wo', rate='250m')
+
+        seconds_to_wait_for_io_operations = 120
+        logging.info(f"Going to sleep for {seconds_to_wait_for_io_operations} seconds")
+        time.sleep(seconds_to_wait_for_io_operations)
+        used_capacity_after_io_operations = temp_helper_file.get_percent_used_capacity()
+        logging.info(f"### used capacity after IO = {used_capacity_after_io_operations}")
+
+        osd_size = storage_cluster.get_osd_size()
+        logging.info("Calling add_capacity function...")
+        result = storage_cluster.add_capacity(osd_size)
+        if result:
+            logging.info("add capacity finished successfully")
+        else:
+            logging.info("add capacity failed")
+
+        # Restart nodes while additional storage is being added
+        logging.info("Restart nodes:")
+        logging.info([n.name for n in node_list])
+        nodes.restart_nodes(nodes=node_list)
+        wait_for_nodes_status(node_names=[node.name for node in node_list])
+        logging.info("Finished restarting the node list")
+
+        used_capacity_after_io_operations = temp_helper_file.get_percent_used_capacity()
+        logging.info(f"### used capacity after node restart = {used_capacity_after_io_operations}")
+
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        pod.wait_for_resource(
+            timeout=300,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=result * 3
+        )
+
+        assert check_osd_pods_after_expansion(osd_pods_before), \
+            "number of osd pods is not as expected"
+        # Don't failed the test for now, just add a warning
+        if not check_total_space_after_expansion(total_space_b4_expansion):
+            logging.warning("Expected capacity mismatch")
+        if not check_osd_tree():
+            logging.warning("Incorrect ceph osd tree format")
+
+        logging.info("Finished verifying add capacity osd storage with node restart")
+        logging.info("Waiting for ceph health check to finished...")
+        ceph_health_check(
+            namespace=config.ENV_DATA['cluster_namespace'], tries=60
+        )
```

Comment:
I suggest referring to PR https://github.com/red-hat-storage/ocs-ci/pull/1751. It's almost covering all the entry and exit criteria. All you need is to add failure condition and verification items. It's better to re-use the code from there for this test and also other Tier4 tests.

Reply:
When the branch 'wip-add-capacity-e_e' will be merged into master I will use the exit criteria from this branch. Currently, we don't want to wait for this PR will be merged into master, cause it may take some time.